### PR TITLE
Update ViewController

### PR DESCRIPTION
**Pull request by fromFrombl4**
add textFields
  add uiswitch
  add constraints

### CODE REVIEW

File: SimpleApp/GalleryCollectionViewCell.swift [hunk 1]
```diff
@@ -1,12 +1,54 @@
...
+    }
+
     let imageView: UIImageView = {
         let imageV = UIImageView()
         imageV.translatesAutoresizingMaskIntoConstraints = false
         return imageV
     }()
 
+    let label: UILabel = {
+        var splashImage: SplashImage?
```
Comment (sergeyduhovich) [inline SimpleApp/GalleryCollectionViewCell.swift]: unused/unset variable

File: SimpleApp/GalleryCollectionViewCell.swift [hunk 2]
```diff
@@ -1,12 +1,54 @@
...
+
+    var splashImage: SplashImage? {
+        didSet {
+            label.text = splashImage?.altDescription
+            label.isHidden = label.text == nil
+            guard let splashImage = splashImage, let urlRegular = URL(string: splashImage.urls.regular) else {
+                imageView.image = nil
+                return
+            }
+            imageView.kf.setImage(with: urlRegular)
```
Comment (sergeyduhovich) [inline SimpleApp/GalleryCollectionViewCell.swift:46]: splashImage/color is not set

File: SimpleApp/ViewController.swift [hunk 3]
```diff
@@ -2,34 +2,124 @@ import UIKit
...
+
+    private var username: String? = nil
+    private var password: String? = nil
+    private var isChecked = false
+
+    @IBAction func saveCredsSwitchChanged() {
+        //update UD isCheckedKey
+        //if OFF
+        // remove userNameKey & passwordKey
+        if saveCredsSwitch.isOn == false {
```
Comment (sergeyduhovich) [inline SimpleApp/ViewController.swift]: `saveCredsSwitch.isOn == false` the same as

`!saveCredsSwitch.isOn`

if no objections on rw swift style guide, use second option

File: SimpleApp/ViewController.swift [hunk 4]
```diff
@@ -2,34 +2,124 @@ import UIKit
...
+        if username == Constants.username && password == Constants.password {
+            //if swiftch ON save user&pass to UD
+            if saveCredsSwitch.isOn {
+                UserDefaults.standard.set(username, forKey: Constants.userNameKey)
+                UserDefaults.standard.set(password, forKey: Constants.passwordKey)
+            }
 
-        for x in 0..<items.count {
-            items[x].image = UIImage(systemName: images[x])
+            let tabBarVC = UITabBarController()
```
Comment (sergeyduhovich) [inline SimpleApp/ViewController.swift]: use separate method for pushing controolers

File: SimpleApp/ViewController.swift [hunk 5]
```diff
@@ -2,34 +2,124 @@ import UIKit
...
+                return
+            }
+            let images = ["house", "star"]
+            for x in 0..<items.count {
+                items[x].image = UIImage(systemName: images[x])
+            }
+            tabBarVC.modalPresentationStyle = .fullScreen
+            present(tabBarVC, animated: true, completion: nil)
+        } else {
+            let alert = UIAlertController(title: "Oops!", message: "Your credentials are wrong", preferredStyle: .alert)
```
Comment (sergeyduhovich) [inline SimpleApp/ViewController.swift]: use separate method for alert

**Review by sergeyduhovich — COMMENTED** [review]
fix lint issues and comments

File: SimpleApp/GalleryCollectionViewCell.swift (hunk 1 above)
Comment (fromFrombl4) [reply SimpleApp/GalleryCollectionViewCell.swift]: fixed

File: SimpleApp/GalleryCollectionViewCell.swift (hunk 2 above)
Comment (fromFrombl4) [reply SimpleApp/GalleryCollectionViewCell.swift:46]: fixed

File: SimpleApp/ViewController.swift (hunk 3 above)
Comment (fromFrombl4) [reply SimpleApp/ViewController.swift]: fixed

File: SimpleApp/ViewController.swift (hunk 4 above)
Comment (fromFrombl4) [reply SimpleApp/ViewController.swift]: fixed

File: SimpleApp/ViewController.swift (hunk 5 above)
Comment (fromFrombl4) [reply SimpleApp/ViewController.swift]: fixed

File: Podfile [hunk 6]
```diff
@@ -6,6 +6,9 @@ target 'SimpleApp' do
   use_frameworks!
 
   pod 'SwiftLint'
+  pod 'Kingfisher'
+  pod 'Hex'
+  pod 'M13Checkbox'
```
Comment (sergeyduhovich) [inline Podfile]: please, remove the unused pod for now

File: SimpleApp/GalleryCollectionViewCell.swift [hunk 7]
```diff
@@ -1,12 +1,53 @@
...
 
+    let label: UILabel = {
+        let labelV = UILabel()
+        labelV.backgroundColor = UIColor(hex: "ff0000")
+        labelV.translatesAutoresizingMaskIntoConstraints = false
+        labelV.numberOfLines = 0
+        return labelV
+    }()
+
+    override func prepareForReuse() { // очистить ячейку до первоначального состояния
```
Comment (sergeyduhovich) [inline SimpleApp/GalleryCollectionViewCell.swift]: please, remove comment

File: SimpleApp/ViewController.swift [hunk 8]
```diff
@@ -2,34 +2,148 @@ import UIKit
...
+        static let userNameKey = "userNameKey"
+        static let passwordKey = "passwordKey"
+        static let isCheckedKey = "isCheckedKey"
+        static let username = "user"
+        static let password = "password"
+    }
+
+    private var username: String?
+    private var password: String?
+    private var isChecked = false
```
Comment (sergeyduhovich) [inline SimpleApp/ViewController.swift]: in this implementation `isChecked` is unused variable, could be removed

File: SimpleApp/ViewController.swift [hunk 9]
```diff
@@ -2,34 +2,148 @@ import UIKit
...
+
+    @IBAction func saveCredsSwitchChanged() {
+        removeNamePasswordIfNeeded()
+    }
 
     override func viewDidLoad() {
         super.viewDidLoad()
+        usernameTextField.delegate = self
+        passwordTextField.delegate = self
+        passwordTextField.isSecureTextEntry = false
```
Comment (sergeyduhovich) [inline SimpleApp/ViewController.swift]: `isSecureTextEntry = false` is default value, assume, `true` is desired behavior

File: SimpleApp/ViewController.swift [hunk 10]
```diff
@@ -2,34 +2,148 @@ import UIKit
...
+            UserDefaults.standard.removeObject(forKey: Constants.userNameKey)
+            UserDefaults.standard.removeObject(forKey: Constants.passwordKey)
+        }
+    }
+
+    private func saveNamePasswordIfNeeded() {
+        let expression = saveCredsSwitch.isOn
+        print(#function + "\(expression)")
+        if expression {
+            username = usernameTextField.text
```
Comment (sergeyduhovich) [inline SimpleApp/ViewController.swift]: `username` already up to date cause `updateUsernameProperty`, might be removed without loss any behavior

File: SimpleApp/ViewController.swift [hunk 11]
```diff
@@ -2,34 +2,148 @@ import UIKit
...
+            password = passwordTextField.text
```
Comment (sergeyduhovich) [inline SimpleApp/ViewController.swift]: same for `password` , might be removed without loss any behavior

File: SimpleApp/ViewController.swift [hunk 12]
```diff
@@ -2,34 +2,148 @@ import UIKit
...
+            for: .editingChanged
+        )
+        restoreNamePasswordIfNeeded()
+        restoreSwitchState()
     }
-    @IBAction func buttonPressed(_ sender: UIButton) {
+
+    private func removeNamePasswordIfNeeded() {
+        let expression = !saveCredsSwitch.isOn
+        print(#function + "\(expression)")
```
Comment (sergeyduhovich) [inline SimpleApp/ViewController.swift]: (optional) all prints might be removed

**Review by sergeyduhovich — CHANGES_REQUESTED** [review]

File: SimpleApp/ViewController.swift (hunk 9 above)
Comment (fromFrombl4) [reply SimpleApp/ViewController.swift]: fixed

File: SimpleApp/ViewController.swift (hunk 12 above)
Comment (fromFrombl4) [reply SimpleApp/ViewController.swift]: fixed

File: SimpleApp/ViewController.swift (hunk 10 above)
Comment (fromFrombl4) [reply SimpleApp/ViewController.swift]: fixed

File: SimpleApp/ViewController.swift (hunk 11 above)
Comment (fromFrombl4) [reply SimpleApp/ViewController.swift]: fixed

**Pull request merged**